app/models/proxy.rb

```diff
@@ -116,6 +116,10 @@ def policies_config=(attr_policies_config)
     super(attr_policies_config.is_a?(String) ? attr_policies_config : attr_policies_config.to_json)
   end
 
+  def find_policy_config_by(name:, version:)
+    policies_config.find { |config| config['name'] == name && config['version'] == version }
```

Comment:
I'm a bit scared about the efficiency of doing it this way + https://github.com/3scale/porta/pull/714/files#diff-f3730ab40abba9f1afcd06dd106439f0R67
Plus, Policy seems to know a bit too much about the internals of Proxy because it does it checking them all from Policy instead of having a class method in Proxy that does it itself, but probably that would be even worse because it could be called for all proxies of the DB.
Another option would be having something like this here in Proxy (and it could be a scope containing the `where` instead):
```ruby
def self.exists?(name:, version:)
  where("policies_config LIKE ?", "%\"name\":\"#{name}\",\"version\":\"#{version}\"%").any?
end
```
The problem of this is we rely even on the order of the fields which we are not checking anywhere that they must be saved in this order, so another option would be:
```ruby
def self.exists?(name:, version:)
  where("policies_config LIKE ?", "%\"name\":\"#{name}\"%").where("policies_config LIKE ?", "%\"version\":\"#{version}\"%").any?
end
```
And then from https://github.com/3scale/porta/pull/714/files#diff-f3730ab40abba9f1afcd06dd106439f0R67 you could do:
```ruby
def in_use?
    account.proxies.exists?(name_was, version_was)
end
```

WDYT @guicassolato ?

Reply:
I thought about that, @Martouta . That was my first design actually, but I never liked the idea of querying the db with a `LIKE` clause on a json content. If it was PostgreSQL only, we could easily cast and enhance the query traversing the json object, but as you know, it is not.

Considering the number of policies that a provider is expected to have (I'm guessing) and the frequency an update or delete operation on a policy will happen, then the current design may not be such a performance killer. I would wait for this to become an issue before breaking our heads too much with a different design.